basic/src/test/java/org/jboss/hal/testsuite/test/configuration/picketlink/federation/SpAttrTestCase.java

```diff
@@ -0,0 +1,99 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.hal.testsuite.test.configuration.picketlink.federation;
+
+import static org.jboss.hal.testsuite.test.configuration.picketlink.federation.UtilFedName.*;
+
+import java.io.IOException;
+
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.dmr.ModelNode;
+import org.jboss.hal.testsuite.category.Shared;
+import org.jboss.hal.testsuite.test.configuration.picketlink.federation.UtilFedInputChecker.InputType;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.OperationException;
+
+/**
+ * Created by pjelinek on Nov 29, 2015
+ */
+@RunWith(Arquillian.class)
+@Category(Shared.class)
+public class SpAttrTestCase extends AbstractFederationTestCase {
+
+    private String
+        federationName = name.getFederation(),
+        spName = name.getSp(),
+        url = name.get(Key.URL);
+
+    private Address spAddress = federationOps.getServiceProviderAddress(federationName, spName);
+
+    @Before
+    public void before() throws IOException, OperationException {
+        setupFederationWithSp();
+        page.navigateToServiceProvider(federationName, spName).switchSubTab(SERVICE_PROVIDER_LABEL);
+    }
+
+    @After
+    public void after() throws IOException, OperationException {
+        cleanupFederation();
+    }
+
+    @Test
+    public void errorPage() throws Exception {
+        edit(InputType.TEXT, spAddress, SP_ERROR_PAGE, new ModelNode("/my_error.jsp")).andVerifySuccess();
+    }
+
+    @Test
+    public void logoutPage() throws Exception {
+        edit(InputType.TEXT, spAddress, SP_LOGOUT_PAGE, new ModelNode("/my_logout.jsp")).andVerifySuccess();
+    }
+
+    @Test
+    public void postBinding() throws Exception {
+        edit(InputType.TEXT, spAddress, SP_POST_BINDING, new ModelNode("false")).andVerifySuccess();
+        edit(InputType.TEXT, spAddress, SP_POST_BINDING, new ModelNode("true")).andVerifySuccess();
+    }
+
+    @Test
+    public void strictPostBinding() throws Exception {
+        edit(InputType.CHECKBOX, spAddress, SP_STRICT_POST_BINDING, new ModelNode(false)).andVerifySuccess();
+        edit(InputType.CHECKBOX, spAddress, SP_STRICT_POST_BINDING, new ModelNode(true)).andVerifySuccess();
+    }
+
+    @Test
+    public void url() throws Exception {
+        edit(InputType.TEXT, spAddress, URL, new ModelNode("")).andVerifyFailure();
```

Comment:
You are trying to detect whether the URL won't be an empty String, shouldn't this check be rather against undefined value?